elyra/cli/pipeline_app.py

```diff
@@ -63,22 +84,30 @@ def _preprocess_pipeline(pipeline_path, runtime, runtime_config):
     if len(pipeline_definition['pipelines']) == 0:
         raise click.ClickException("Pipeline has zero length 'pipelines' field.")
 
+    primary_pipeline_key = pipeline_definition['primary_pipeline']
+    primary_pipeline = None
     try:
         for pipeline in pipeline_definition["pipelines"]:
+            if pipeline['id'] == primary_pipeline_key:
+                primary_pipeline = pipeline
             for node in pipeline["nodes"]:
                 if 'filename' in node["app_data"]:
                     abs_path = os.path.join(pipeline_dir, node["app_data"]["filename"])
                     node["app_data"]["filename"] = abs_path
 
-        # NOTE: The frontend just set the info for first pipeline, but shouldn't it
-        # search for the primary pipeline and set that?
-        # Setting `pipeline_definition["pipelines"][0]` for consistency.
-        pipeline_definition["pipelines"][0]["app_data"]["name"] = pipeline_name
-        pipeline_definition["pipelines"][0]["app_data"]["runtime"] = runtime
-        pipeline_definition["pipelines"][0]["app_data"]["runtime-config"] = runtime_config
     except Exception as e:
         raise click.ClickException(f"Error pre-processing pipeline: \n {e}")
 
+    if not _validate_pipeline_runtime(primary_pipeline, runtime):
+        raise click.ClickException("Runtime mismatch - Trying to run a runtime specific pipeline"
+                                   " using a configuration for a different runtime.")
```

Comment:
@ptitzler Could you please suggest a better message here...

Reply:
How about something like this, with `{}` being placeholders for the appropriate values.

```
This pipeline requires a {Kubeflow Pipelines | Apache Airflow} runtime configuration. The specified configuration {rtc_name} is for {Kubeflow Pipelines | Apache Airflow}. 
```